presentation/src/main/java/bg/government/virusafe/app/utils/validators/EmbgValidator.kt

```diff
@@ -0,0 +1,116 @@
+@file:Suppress("MagicNumber")
+package bg.government.virusafe.app.utils.validators
+
+import bg.government.virusafe.app.utils.empty
+import com.upnetix.applicationservice.registration.model.Gender
+import org.threeten.bp.LocalDate
+import org.threeten.bp.temporal.ChronoUnit
+
+/**
+ * Validate EMBG.
+ */
+private const val EMBG_SKIP_FACTOR = 6
+private const val EMBG_MOD = 11
+private const val EMBG_BASE = 11
+
+class EmbgValidator : PersonalIdValidator {
+
+	private val embgWeights = listOf(7, 6, 5, 4, 3, 2)
+	private var personalNumber: String? = null
+	private var isValid = false
+
+	override var years: Long? = null
+		private set
+
+	override var gender: String? = null
+		private set
+
+	override fun isValidPersonalId(): Boolean = isValid
+
+	override fun initPersonalNumber(personalNumber: String?) {
+		this.personalNumber = personalNumber
+		isValid = isValidPersonalNumber()
+		if (!isValid) {
+			years = null
+			gender = null
+		}
+	}
+
+	override fun clearAll() {
+		personalNumber = null
+		isValid = false
+		years = null
+		gender = null
+	}
+
+	@Suppress("ReturnCount")
+	private fun isValidPersonalNumber(): Boolean {
+		if (personalNumber == null || personalNumber == String.empty) {
+			return true
+		} else if (personalNumber!!.length != 13) {
```

Comment:
Avoid using the not-null assertion operator (!!) use safe call operator instead (?.)